tests/rptest/scale_tests/cloud_storage_compaction_test.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import time
+
+from rptest.clients.kafka_cli_tools import KafkaCliTools
+from rptest.clients.rpk import RpkTool, RpkException
+from rptest.clients.types import TopicSpec
+from rptest.services.cluster import cluster
+from rptest.services.redpanda import RedpandaService
+from rptest.services.redpanda import SISettings
+from rptest.tests.end_to_end import EndToEndTest
+from rptest.util import wait_until
+from rptest.services.kgo_verifier_services import KgoVerifierRandomConsumer
+
+LOCAL_CONFIGURATION = {
+    "partition_amount": 3,
+    "segment_size": 1024 * 100,
```

Comment:
for compacted segments you may need to set `compacted_log_segment_size` instead of this property.